File: app/src/main/java/activitytest/example/com/roomdemo/main/database/ConfigurationDao.java
package activitytest.example.com.roomdemo.main.database;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

/**
 * configuration的dao层
 */
@Dao
public interface ConfigurationDao {

    @Insert
    void insertConfiguration(Configuration[] configuration);

    @Query ( "SELECT * FROM CONFIGURATION" )
    LiveData<Configuration> queryConfiguration();

    @Update
    void updateConfiguration(Configuration[] configuration);
}

File: app/src/main/java/activitytest/example/com/roomdemo/main/database/dao/ConfigurationDao.kt
package activitytest.example.com.roomdemo.main.database.dao

import activitytest.example.com.roomdemo.main.database.entity.Configuration
import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update

/**
 * configuration的dao层
 */
@Dao
interface ConfigurationDao {
    @Insert
    suspend fun insertConfiguration(configuration: Configuration)

    @Query("SELECT * FROM CONFIGURATION")
     fun queryConfiguration(): LiveData<Configuration?>?

    @Update
    suspend fun updateConfiguration(configuration: Configuration)

}